### ISSUE TITLE
Update walkthrough for selecting the feature type before drawing

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#6025]
#6010 broke most chapters of the walkthrough. These must be updated before release.

Affected chapters (if we want to keep these divisions):
- [ ] Points
- [ ] Lines
- [ ] Areas
- [ ] Buildings

**Comment by bhousel** [comment]
Yeah this will be a lot of work, both for us and for translators, so I want to make sure we are really solid on how the new ribbon UI will work before we start changing the walkthrough.  So I'm going to insist that we take time (maybe a week or two) to make it amazing first and do a lot of editing with it, and show it to people.  I'm glad that it's in preview now and people can start using it today.

Also, while redoing the walkthrough, we could consider promisifying the whole thing, and maybe swap out our [`q`](https://www.npmjs.com/package/q) dependency for something more modern like [`bluebird`](https://www.npmjs.com/package/bluebird) / native promises in browsers that support them.. It might make the walkthrough content easier to work with (and we don't need to wait to start on that).

**Comment by mmd-osm** [comment]
Does it make sense to upon some new issues at this time, or should I wait some time?

I tried adding a _bus stopping location_,  which maps to a "vertex" object type. However, the UI only shows some crosshair cursor but won't let me create any vertex object.

**Comment by bhousel** [comment]
> Does it make sense to open some new issues at this time, or should I wait some time?

Yes - please open issues for things like this that you find confusing 👍

**Comment by bhousel** [comment]
BTW, I have a real problem with the word "vertex".  I've tested iD with many newbies and they really hate that word.

**Comment by manfredbrandl** [comment]
I share the problem with the wort "vertex", i find it confusing enough to have point and node. BTW: Vertex is according to transifex used in "presets" twice and in "core" four times, so translators wont cry if vertex is replaced by node maybe.

**Comment by SilentSpike** [comment]
To my mind vertex/point/node are all synonymous. Do they mean different things in iD?

**Comment by quincylvania** [comment]
> To my mind vertex/point/node are all synonymous. Do they mean different things in iD?

@SilentSpike Generally in iD, "node" means any OSM node, "point" means a node that's not part of any way, and "vertex" means a point that is part of some way.

> I tried adding a bus stopping location, which maps to a "vertex" object type. However, the UI only shows some crosshair cursor but won't let me create any vertex object.

@mmd-osm This is since it's making sure "vertex" features are added to existing ways, not as arbitrary points. Otherwise when you added the point it wouldn't match the preset. With #5960 we will tell the user this explicitly. We will also need error feedback so the use will know why nothing happens when you click.

> BTW, I have a real problem with the word "vertex". I've tested iD with many newbies and they really hate that word.

@bhousel  Huh, I didn't realize it was a big deal. I was using it to differentiate the presets that can be added to ways from those that can't. I'm fine with finding another way to present this difference, or possibly dropping the distinction.

> Yeah this will be a lot of work, both for us and for translators, so I want to make sure we are really solid on how the new ribbon UI will work before we start changing the walkthrough. So I'm going to insist that we take time (maybe a week or two) to make it amazing first and do a lot of editing with it, and show it to people.

I totally agree. The new workflow is a major change and it's worth taking our time to get it right. I appreciate everyone's willingness to thoughtfully consider the new features and give feedback! ✌️

**Comment by bhousel** [comment]
>> BTW, I have a real problem with the word "vertex". I've tested iD with many newbies and they really hate that word.
>
> @bhousel Huh, I didn't realize it was a big deal. I was using it to differentiate the presets that can be added to ways from those that can't. I'm fine with finding another way to present this difference, or possibly dropping the distinction.

Yes, it probably sounds silly but I spent actual months refining the tutorial for the broadest audience possible, and tested it with several people who had never edited OSM before or had any familiarity with mapmaking.

Most non-technical people hate the word _node_, and _vertex_ is even worse.  Like "close-the-laptop-this-is-not-for-me" bad.  

The words we use in iD are super important.  Calling vertices "Other" was not great either, but it was a purposeful decision.  iD is a mature project, and almost everything in it was done that way for some reason.  This project exists mostly to make everyone feel confident that they can be a mapper.  Not to add the most stuff or the correctest stuff to OSM.

I'm really hopeful that we can retain iD's approachability and ease of use, and still experiment with the UI to help intermediate mappers be more efficient.

**Comment by mmd-osm** [comment]
I was scratching my head when I first came across _vertex_ in the iD sources, as no other part of the OSM ecosystem uses this term really. Everyone talks in nodes, ways and relations. Even "area" is a bit special, as we still don't have an area data type in the OSM data model yet. 

I still vaguely remember this bit here, which translates iD to the "rest of the world" ;)

https://github.com/openstreetmap/iD/blob/764acea67fcac7b8df70b59e45440b0ddd75045f/build_data.js#L435-L439

**Comment by manfredbrandl** [comment]
I am doing translation into German: point is Punkt, vertex/node x are Knoten now. I think about translating these three with Punkt only, because everybody will easily understand that. I think we do not need extra word for all points, points which are part of a way and point with are standalone, one name for all should be enough. Everybody agree?

**Comment by quincylvania** [comment]
> I think we do not need extra word for all points, points which are part of a way and point with are standalone, one name for all should be enough. Everybody agree?

This sounds fine for iD, yes.

**Comment by k-yle** [comment]
I'm going to close this issue, since it's an issue in the v3 prototype version, which is (sadly) abandoned and [no longer deployed](https://github.com/openstreetmap/iD/commit/36f4eeb7e24ba3ffd16f0d48780f3c56d39f2bd1).
